.dagger-ci/daggerci/tests/test_docker_compose.py

```diff
@@ -0,0 +1,97 @@
+# pylint: disable=missing-function-docstring
+# pylint: disable=missing-module-docstring
+# pylint: disable=too-many-arguments
+# mypy: disable-error-code="import, no-untyped-def"
+
+import os
+from contextlib import nullcontext as does_not_raise
+
+import dagger
+import pytest
+from lib.docker_compose import (
+    DockerCompose,
+    DockerComposeMissingElement,
+    DockerComposeValidate,
+    select,
+)
+
+
+@pytest.mark.parametrize(
+    "heap, needle, expected, expectation",
+    [
+        (["a", "b"], "a", "a", does_not_raise()),
+        (["a", "b"], "b", "b", does_not_raise()),
+        (["a", "b"], None, "a", does_not_raise()),
+        (["a", "b"], "c", "a", pytest.raises(ValueError)),
+    ],
+)
+def test__select(heap, needle, expected, expectation):
+    with expectation:
+        assert select(heap=heap, needle=needle) == expected
+
+
+def test__docker_compose_broken(tmpdir, create_file, docker_compose_file_broken):
+    compose_file = os.path.join(tmpdir, "compose.yaml")
+    create_file(path=compose_file, content=docker_compose_file_broken)
+    with pytest.raises(DockerComposeValidate):
+        DockerCompose(path=compose_file)
+
+
+def test__docker_compose(tmpdir, create_file, docker_compose_file_complex):
+    compose_file = os.path.join(tmpdir, "compose.yaml")
+    create_file(path=compose_file, content=docker_compose_file_complex)
+    mydockercompose = DockerCompose(path=compose_file)
+
+    assert mydockercompose.get_top_elements() == ["services"]
+
+    assert mydockercompose.__select_top_element__() == "services"
+    assert mydockercompose.__select_top_element__("services") == "services"
+    with pytest.raises(DockerComposeMissingElement):
+        mydockercompose.__select_top_element__("")
+
+    assert mydockercompose.get_dockerfiles() == [
+        "coreboot_4.19",
+        "coreboot_4.20",
+        "edk2",
+        "meh",
```

Comment:
I suppose the `meh` and `meh2` files are still placeholders?

Reply:
I just could not think of anything smarter. Open to suggestions.

Reply:
I don't have an issues with funny names as long as they are expressive. And from the outside I can not tell what `meh` and `meh2` are used for. So either choose a boring but descriptive word like test, stub, dummy, unknown, no-op etc or choose a funnier name which at least makes immediately clear what the scope/usecase is

Reply:
dummy it is, out of funny today